### ISSUE TITLE
beforeUpdate execute twice

### ISSUE THREAD

**Issue by devmark** [overlookmotel/sequelize-hierarchy#43]
`SequelizeUniqueConstraintError` because beforeUpdate run twicw
1. first time trigger : updated parentId
2. second time trigger: updated level by first time beforeUpdate
```
 return employee.updateAttributes({
                        managerId: 1
                    });
```

```
      hierarchy: {
            as: 'manager',
            foreignKey: 'managerId',
            throughKey: 'employeeId',
            levelFieldName: 'depth',
            throughSchema: 'fin',
            camelThrough: true
        }
```
already tried `options.validate = false`;

sequelize: `3.14.0`
postgresql: 

should add a condition:

```
if(item._changed[hierarchy.levelFieldName] === true) return;
```

any idea?

**Comment by overlookmotel** [comment]
Thanks for raising this issue. I'll investigate.

To make it easier for me, can you please post the full definition of the `Employee` model?

**Comment by devmark** [comment]
Thanks for reply. 
using: postgresql & sequelize(^3.14.0)
model:
```
    var EmployeeModel = sequelize.define('Employee', {
        id: {
            type: DataTypes.INTEGER,
            allowNull: false,
            autoIncrement: true,
            primaryKey: true
        },
        lastName: {
            type: DataTypes.STRING,
            allowNull: true
        },
        firstName: {
            type: DataTypes.STRING,
            allowNull: true
        },
        managerId: {
            type: DataTypes.INTEGER,
            allowNull: true
        },
        depth: {
            type: DataTypes.INTEGER,
            allowNull: true
        }
    }, {
        charset: 'utf8',
        schema: 'fin',
        validate: false,
        hierarchy: {
            as: 'manager',
            foreignKey: 'managerId',
            throughKey: 'employeeId',
            levelFieldName: 'depth',
            throughSchema: 'hr',
            camelThrough: true
        }
    });
```

action:
```
        return sequelize.transaction().then(function (t) {

            return EmployeeModel.create({
                userId: user.id,
                appId: app.id,
                firstName: req.body.firstName,
                lastName: req.body.lastName,
                joinedDate: new Date()
            }, {transaction: t}).then(function (employee) {
                t.commit().then(function () {
                    return employee.updateAttributes({
                        managerId: 1
                    });
                }).then(function () {
                       
                    });
            }).catch(function (err) {
                t.rollback();
            });
        });
```

**Comment by overlookmotel** [comment]
Great thanks. OK, I'll look into it. It may be while until I can get to it though - am flat out on another project. Why does work always have to get in the way of good times? ;)

**Comment by devmark** [comment]
Happy weekend
Cheer :+1:

**Comment by overlookmotel** [comment]
@devmark It's taken me a long time, but I finally looked into this.

I can't reproduce your issue. The `beforeUpdate` hook does not run an `update` command so I don't think the problem can be that `beforeUpdate` is triggered twice.

Are you sure the `SequelizeUniqueConstraintError` isn't caused by something simpler, like `managerId: 1` is pointing to a non-existent record?

Other possibility is that the malfunction was caused by a lack of schemas support in `sequelize-hierarchy`. This has now been fixed in v0.6.0.

I'm going to close this issue for now, but feel free to re-open if you have the problem reoccurring with v0.6.0.
